editor/src/components/canvas/dom-walker-caching.spec.browser2.tsx

```diff
@@ -199,43 +206,49 @@ describe('Dom-walker Caching', () => {
     expect(simpleStringifyActions(renderResult.getRecordedActions())).toEqual(`[
   SET_ELEMENTS_TO_RERENDER,
   SET_CANVAS_FRAMES,
+  SAVE_DOM_REPORT,
   MERGE_WITH_PREV_UNDO: [
     UPDATE_FROM_WORKER
   ],
   SAVE_DOM_REPORT,
-  SAVE_DOM_REPORT,
   SET_CANVAS_FRAMES,
+  SAVE_DOM_REPORT,
   MERGE_WITH_PREV_UNDO: [
     UPDATE_FROM_WORKER
   ],
-  SAVE_DOM_REPORT,
   SAVE_DOM_REPORT
 ]`)
 
     expect(saveDomReportActions.length).toBe(4)
 
-    expect(saveDomReportActions[1].invalidatedPaths).toEqual(['storyboard-entity/scene-2-entity'])
-    expect(saveDomReportActions[1].cachedPaths).toEqual([
-      EP.fromString('storyboard-entity/scene-1-entity/app-entity:app-outer-div/card-instance'),
-      EP.fromString('storyboard-entity/scene-1-entity/app-entity:app-outer-div'),
-      EP.fromString('storyboard-entity/scene-1-entity/app-entity'),
-      EP.fromString('storyboard-entity/scene-1-entity'),
+    expect(saveDomReportActions[1].invalidatedPaths).toEqual([
+      'storyboard-entity',
+      'storyboard-entity/scene-1-entity/app-entity:app-outer-div',
+      'storyboard-entity/scene-1-entity/app-entity:app-outer-div/card-instance:card-outer-div',
+      'storyboard-entity/scene-2-entity/same-file-app-entity:same-file-app-div',
     ])
+    expect(saveDomReportActions[1].cachedPaths).toEqual([])
 
     expect(saveDomReportActions[2].invalidatedPaths).toEqual([
+      'storyboard-entity/scene-2-entity',
       'storyboard-entity',
       'storyboard-entity/scene-1-entity/app-entity:app-outer-div',
       'storyboard-entity/scene-1-entity/app-entity:app-outer-div/card-instance:card-outer-div',
       'storyboard-entity/scene-2-entity/same-file-app-entity:same-file-app-div',
     ])
-    expect(saveDomReportActions[2].cachedPaths).toEqual([])
-
-    expect(saveDomReportActions[3].invalidatedPaths).toEqual(['storyboard-entity/scene-2-entity'])
-    expect(saveDomReportActions[3].cachedPaths).toEqual([
+    expect(saveDomReportActions[2].cachedPaths).toEqual([
       EP.fromString('storyboard-entity/scene-1-entity/app-entity:app-outer-div/card-instance'),
       EP.fromString('storyboard-entity/scene-1-entity/app-entity:app-outer-div'),
```

Comment:
As mentioned, this path shouldn't appear in both the `cachedPaths` and `invalidatedPaths` for the same dom walking result, so this definitely needs some investigating

Reply:
When a selective re-render is triggered, it can end up having paths that differ to what ends up in `invalidatedPaths`, which when results in the unchanged elements ending up also in `cachedPaths`.